packages/flutter_tools/lib/src/build_system/depfile.dart

```diff
@@ -82,7 +78,7 @@ class DepfileService {
 
   void _writeFilesToBuffer(List<File> files, StringBuffer buffer) {
     for (final File outputFile in files) {
-      if (_platform.isWindows) {
+      if (_fileSystem.path.style.separator == r'\') {
         // Foward slashes and spaces in a depfile have to be escaped on windows.
```

Comment:
Right, backslashes :) 

We escape spaces on every platform, but the backslashes need to be escaped on Windows specifically because of how we pass this info to gradle.